### PR TITLE
[loki-distributed] Change memcached_client addresses for query_range

### PR DESCRIPTION
**Pull request by patsevanton**
Fix error https://github.com/grafana/helm-charts/issues/1127: 
``
failed to lookup SRV records" host=_memcached-client._tcp.loki-loki-distributed-memcached-frontend.loki.svc.cluster.local err="lookup _memcached-client._tcp.loki-loki-distributed-memcached-frontend.loki.svc.cluster.local on 10.97.128.2:53: no such host
``

Signed-off-by: Anton Patsev <patsev.anton@gmail.com>

When value.yaml:
```
loki:
  schemaConfig:
    configs:
      - from: 2020-09-07
        store: boltdb-shipper
        object_store: s3
        schema: v11
        index:
          prefix: loki_index_
          period: 24h

  storageConfig:
    boltdb_shipper:
      shared_store: s3
    aws:
      endpoint: xxxx
      s3forcepathstyle: true

memcachedFrontend:
  enabled: true
```

### CODE REVIEW

File: charts/loki-distributed/values.yaml [hunk 1]
```diff
@@ -162,8 +162,12 @@ loki:
         cache:
           {{- if .Values.memcachedFrontend.enabled }}
           memcached_client:
-            addresses: dnssrv+_memcached-client._tcp.{{ include "loki.memcachedFrontendFullname" . }}.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}
             consistent_hash: true
+            host: {{ include "loki.memcachedFrontendFullname" . }}
+            max_idle_conns: 16
+            service: memcached-client
+            timeout: 500ms
```
Comment (z0rc) [inline charts/loki-distributed/values.yaml:169]: Don't change default values.

File: charts/loki-distributed/values.yaml [hunk 2]
```diff
@@ -162,8 +162,12 @@ loki:
         cache:
           {{- if .Values.memcachedFrontend.enabled }}
           memcached_client:
-            addresses: dnssrv+_memcached-client._tcp.{{ include "loki.memcachedFrontendFullname" . }}.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}
             consistent_hash: true
+            host: {{ include "loki.memcachedFrontendFullname" . }}
+            max_idle_conns: 16
```
Comment (z0rc) [inline charts/loki-distributed/values.yaml:167]: This is default, no point of setting it here.

File: charts/loki-distributed/values.yaml [hunk 3]
```diff
@@ -162,8 +162,12 @@ loki:
...
+            service: memcached-client
+            timeout: 500ms
+            update_interval: 1m
```
Comment (z0rc) [inline charts/loki-distributed/values.yaml:170]: This is already default. No point of setting it here.

File: charts/loki-distributed/values.yaml [hunk 4]
```diff
@@ -162,8 +162,12 @@ loki:
         cache:
           {{- if .Values.memcachedFrontend.enabled }}
           memcached_client:
-            addresses: dnssrv+_memcached-client._tcp.{{ include "loki.memcachedFrontendFullname" . }}.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}
             consistent_hash: true
+            host: {{ include "loki.memcachedFrontendFullname" . }}
+            max_idle_conns: 16
+            service: memcached-client
```
Comment (z0rc) [inline charts/loki-distributed/values.yaml:168]: `service` is used only when `host` is empty.

**Review by z0rc — CHANGES_REQUESTED** [review]
Check `memcached_client` block at https://grafana.com/docs/loki/latest/configuration/#cache_config

File: charts/loki-distributed/values.yaml (hunk 1 above)
Comment (patsevanton) [reply charts/loki-distributed/values.yaml]: Remove default values.

File: charts/loki-distributed/values.yaml (hunk 3 above)
Comment (patsevanton) [reply charts/loki-distributed/values.yaml]: Remove default values.

File: charts/loki-distributed/values.yaml (hunk 2 above)
Comment (patsevanton) [reply charts/loki-distributed/values.yaml]: Remove default values.

**Comment by patsevanton** [comment]
@z0rc Thanks for review.

**Comment by patsevanton** [comment]
@unguiculus and @Whyeasy please review PR.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**